src/ucp/wireup/wireup.c

```diff
@@ -738,7 +736,19 @@ ucs_status_t ucp_wireup_init_lanes(ucp_ep_h ep, const ucp_ep_params_t *params,
     status = ucp_wireup_select_lanes(ep, params, ep_init_flags, address_count,
                                      address_list, addr_indices, &key);
     if (status != UCS_OK) {
-        goto err;
+        if (ep->flags & UCP_EP_FLAG_USED) {
+            failed_lane = 0; /* lanes aren't selected but need to invoke err
```

Comment:
why lane number doesn't matter, you are assuming it exists and set it to failed?
maybe ucp_worker_set_ep_failed should handle failed_lane=NULL to treat all lanes as failed?